### PR TITLE
Search: Fix currency code

### LINKED ISSUE #24986: Jetpack Search upsell page is not showing currency code
### Impacted plugin

Search

### Steps to Reproduce

1. Go to a site with only the Search plugin
2. Connect site, but don't purchase any plan
3. Navigate to `/wp-admin/admin.php?page=jetpack-search`

### A clear and concise description of what you expected to happen.

The pricing should be shown in a supported local currency

e.g.

image

### What actually happened

It's not showing in local currency:

image

### Other information

I think it's because the currency code is not passed in with the right name:

https://github.com/Automattic/jetpack/blob/69c71ea6a88f4348cd8cfaf1f492d541e6396f57/projects/packages/search/src/dashboard/components/pages/upsell-page.jsx#L82

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Workaround details

_No response_

### Operating System

Other

### OS Version

_No response_

### Browser

Other / Not applicable

### Browser Version(s)

_No response_

### PR DESCRIPTION
**Pull request by dognose24**
Fixes #24986 

#### Changes proposed in this Pull Request:
Modify prop key from `pricingCurrencyCode` to `currencyCode` for correct currency code in component `PricingCard`.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. Go to a site with only the plugin `Jetpack Search` installed
2. Connect the site to Jetpack, but don't purchase any plan
3. Navigate to URL `/wp-admin/admin.php?page=jetpack-search`
4. Ensure the pricing displaying with the local currency symbol for the user

|  Before   | After  |
|  ----  | ----  |
| b | f |

### CODE REVIEW

File: projects/packages/search/changelog/fix-currency-code [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fixed
+
+Change prop key to component for currency code
```
Comment (kangzj) [inline projects/packages/search/changelog/fix-currency-code]:
```suggestion
Fixed the currency code missing issue for the upsell page
```
Would it be better if we describe it in a higher level? 😄

File: projects/packages/search/src/class-package.php [hunk 2]
```diff
@@ -11,7 +11,7 @@
  * Search package general information
  */
 class Package {
-	const VERSION = '0.16.0';
+	const VERSION = '0.16.1-alpha';
```
Comment (kangzj) [inline projects/packages/search/src/class-package.php:14]: You learnt from the checking error and bumped the version yippee 🎉

File: projects/packages/search/src/dashboard/components/pages/upsell-page.jsx [hunk 3]
```diff
@@ -79,7 +79,7 @@ export default function UpsellPage( { isLoading = false } ) {
 										onCtaClick={ sendToCart }
 										priceAfter={ priceAfter }
 										priceBefore={ priceBefore }
-										pricingCurrencyCode={ priceCurrencyCode }
+										currencyCode={ priceCurrencyCode }
```
Comment (kangzj) [inline projects/packages/search/src/dashboard/components/pages/upsell-page.jsx:82]: Nice fix ❤️ 
This was on me, thanks for fixing it 👍

File: projects/packages/search/changelog/fix-currency-code (hunk 1 above)
Comment (dognose24) [reply projects/packages/search/changelog/fix-currency-code]: Yes, the changelog no needs to be detailed as commit messages!

**Review by kangzj — APPROVED** [review]
It works as described and looks good to me ❤️

**Pull request merged**